app/src/main/java/com/jHerscu/clearskies/ui/prefs/PreferencesScreen.kt

```diff
@@ -1 +1,267 @@
+            PreferenceCarouselItem(
+                sortMethod = stringResource(id = carouselItems[item].titleRes)
+            ) {
+                viewModel.passIntent(Intent.UpdateSortOrderClick(carouselItems[item].name))
+            }
+        }
+        PrimaryElevatedCard(
+            columnModifier = Modifier.padding(Padding.MEDIUM.dp)
+        ) {
+            LazyVerticalGrid(
+                modifier = Modifier.fillMaxSize(),
+                columns = if (windowSizeClass.widthSizeClass != WindowWidthSizeClass.Compact) {
+                    GridCells.Fixed(2)
+                } else {
+                    GridCells.Fixed(1)
+                },
+                horizontalArrangement = Arrangement.Start,
+                verticalArrangement = Arrangement.Top
+            ) {
+                val prefGroups = viewState.value.prefGroups
+                items(
+                    items = prefGroups,
+                    key = { it.name } // TODO(jherscu): Declare localized string as key in case of language pref change?
+                ) { prefGroup ->
+                    GroupedPreferencesCard(
+                        modifier = Modifier
+                            .then(
+                                if (prefGroup != prefGroups.last()) {
+                                    Modifier.padding(bottom = Padding.STANDARD.dp)
+                                } else {
+                                    Modifier
+                                }
+                            ),
+                        isColumn = isColumn,
+                        title = { Text(text = stringResource(id = prefGroup.titleRes)) }
+                    ) {
+                        val weightModifier = Modifier.then(
+                            if (isColumn) {
+                                Modifier.fillMaxWidth()
+                            } else {
+                                Modifier.weight(1f)
+                            }
+                        )
+                        when (prefGroup) {
+                            PreferenceGroup.THEME -> {
+                                PreferenceSwitchComponent(
+                                    isInColumn = isColumn,
+                                    modifier = weightModifier,
+                                    title = stringResource(id = R.string.dynamic_theming),
+                                    optionIsSelected = viewState.value.userPrefs.dynamicThemingOn,
+                                    setOption = { viewModel.passIntent(Intent.DynamicThemingModeClick(it)) }
+                                )
+                                Spacer(modifier = Modifier.size(Padding.MEDIUM.dp))
+                                PreferenceRadioComponent(
+                                    modifier = weightModifier,
+                                    title = stringResource(id = R.string.lock_theme),
+                                    options = LockedThemePref.entries.toPersistentList(),
```

Comment:
Upgraded kotlin version to access Enum entries so I don't have to add helper function to each enum for matching